### PR TITLE
Two handlers, Tests with Mock Events and Mock Blockchain

### PR DESCRIPTION
**Pull request by jose-compu**
### Purpose for this PR

Idem.

### CODE REVIEW

**Comment by jose-compu** [comment]
All checks have passed.

File: scripts/sql/create_tables.sql [hunk 1]
```diff
@@ -1,34 +1,43 @@
 -- Delegatees are account that receive delegation of voting power from Delegators
 CREATE TABLE delegation
 (
-    token_address     CHAR(20),
-    delegator_address CHAR(20),
-    delegatee_address CHAR(20),
+    network          INT,
+    token_address     CHAR(64),
```
Comment (CjHare) [inline scripts/sql/create_tables.sql:5]: issue (blocking): no `token_address` anymore.

File: src/abi.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
...t","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"symbol","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"totalSupply","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint256","name":"snapshotId","type":"uint256"}],"name":"totalSupplyAt","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"recipient","type":"address"},{"internalType":"uint256","name":"amount","type":"uint256"}],"name":"transfer","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"sender","type":"address"},{"internalType":"address","name":"recipient","type":"address"},{"internalType":"uint256","name":"amount","type":"uint256"}],"name":"transferFrom","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"nonpayable","type":"function"}]'
```
Comment (CjHare) [inline src/abi.ts:2]: issue (blocking): Typechain.

_Follow the standard as defined in https://github.com/windranger-io/windranger-solidity-template and use TypeChain, there should no abi files_

_You can use either dependencies on NPM deployable, other Git repos or local contract interface files_

File: src/db/postgres.ts [hunk 3]
```diff
@@ -58,6 +58,23 @@ async function query(dbConfig: ClientConfig, sql: string): Promise<void> {
     await db.end()
 }
 
+async function queryWithParams(
+    dbConfig: ClientConfig,
+    sql: string,
+    params?: any[]
```
Comment (CjHare) [inline src/db/postgres.ts:64]: question: how does this pass linting, there's rule about not allowing `any`?

File: src/handler/get-voting-power.ts [hunk 4]
```diff
@@ -4,8 +4,26 @@ import {
 } from 'aws-lambda/trigger/api-gateway-proxy'
 import {Context} from 'aws-lambda'
 import {log} from '../../config/logging'
-import {Database} from '../db/database'
 import {throwError} from '../error'
+import {getCombinedVotingPowerOf} from '../blockchain'
+import {getCombinedVotingPowerOfMock} from '../../test/blockchainMock'
+
+export const getPriorVotingPowerOfImpl = async (
```
Comment (CjHare) [inline src/handler/get-voting-power.ts:11]: suggestion: use function syntax.

```
export function getPriorVotingPowerOfImpl( walletAddress: string,
    blockNumber: number,
    useMocked: boolean)::Promise<number>
```

**Review by CjHare — CHANGES_REQUESTED** [review]
Use Typechain and hardhat instead of ABI files and the local blockchain.

There should be nothing in the code that defines any network.

File: src/handler/get-voting-power.ts (hunk 4 above)
Comment (jose-compu) [reply src/handler/get-voting-power.ts:11]: ok, anon function is not very readble.

File: src/db/postgres.ts (hunk 3 above)
Comment (jose-compu) [reply src/db/postgres.ts:64]: I added some exceptions, seems to be hard to parse the JSON in TypeScript because is polymorphic.

File: src/abi.ts (hunk 2 above)
Comment (jose-compu) [reply src/abi.ts:2]: gotcha

File: scripts/sql/create_tables.sql (hunk 1 above)
Comment (jose-compu) [reply scripts/sql/create_tables.sql:5]: and no network.
